tools/SourceKit/lib/SwiftLang/SwiftEditor.cpp

```diff
@@ -1049,8 +1059,11 @@ struct SwiftEditorDocument::Implementation {
 
   Implementation(StringRef FilePath, SwiftLangSupport &LangSupport,
                  CodeFormatOptions options)
-    : LangSupport(LangSupport), FilePath(FilePath), FormatOptions(options) {
-      SemanticInfo = new SwiftDocumentSemanticInfo(FilePath, LangSupport);
+      : ASTMgr(LangSupport.getASTManager().shared_from_this()),
```

Comment:
If not convenient to change `getASTManager()` to return the shared_ptr, I recommend to add a `getASTManagerPtr()` method and use it here. Generally, I prefer to only use `shared_from_this()` from inside the class itself, not its users.

Reply:
Turns out it's easy to just return `shared_ptr`, so I fixed it that way.  I was hoping I could prevent this statically by using private inheritance from `std::enable_shared_from_this`, but that doesn't work because the whole mechanism relies on `shared_ptr`'s constructor being able to see the inheritance from `enable_shared_from_this`.  Oh well.